client/src/main.rs

```diff
@@ -1268,11 +1306,13 @@ fn main() {
                                 Some(sig) => get_pair_from_str(sig),
                                 None => panic!("please specify --signer.")
                             };
-                            let mut api = api.set_signer(signer.clone().into());
+                            let mut api = api;
+                            let signer = ParentchainExtrinsicSigner::new(sr25519_core::Pair::from(signer.clone()));
+                            api.set_signer(signer.clone());
```

Comment:
Isn't signer already a `sr25519::Pair`?

Reply:
Almost, it is `AppPair` instead of `Pair`.